src/app/courses/courses/courses.component.ts

```diff
@@ -13,12 +14,12 @@ export class CoursesComponent implements OnInit {
   @ViewChild(CoursesListComponent, {static: false})
   private coursesList: CoursesListComponent;
 
-  constructor(private filter: FilterByTitlePipe) { }
+  constructor(private filter: FilterByTitlePipe, private coursesService: CoursesService) { }
 
   ngOnInit() {
   }
 
   searchCourseByTitle(event: string) {
-    this.coursesList.coursesItems = this.filter.transform(this.coursesList.getCoursesList(), event);
+    this.coursesList.coursesItems = this.filter.transform(this.coursesService.getList(), event);
```

Comment:
somehow I missed it first time, but it's really bad to manipulate children's internal properties from parent component, it would be much better if child will accept list of courses as an input from parent, or you can store filter value in CoursesService and getList will return list of curses after filtering them

in current implementation there is a bug: you filtering courses as a result you can see part of them, then you are removing one of them, courses list will call coursesService.getList() and it will return full list without removed one item, not taking search string into account